### ISSUE TITLE
TypeError: splice is not a function

### ISSUE THREAD

**Issue by leekelleher** [umco/umbraco-stacked-content#2]
**Ignore this until I've gathered more info** :neckbeard:

I'm seeing an intermittent error in my JS console...

    TypeError: $scope.model.value.splice is not a function

Which corresponds to this line.. [stackedcontent.controllers.js#L93](https://github.com/umco/umbraco-stacked-content/blob/develop/src/Our.Umbraco.StackedContent/Web/UI/App_Plugins/StackedContent/js/stackedcontent.controllers.js#L93)

So far I've only seen this happen on the "Single Item" mode ones.
... and only noticed it on those set as a mandatory field.

---

I'll investigate further and update back on here.

**Comment by leekelleher** [comment]
After a while of seeing this error intermittently, I've found a way to reproduce it.

1. On a doctype, set any field as mandatory
2. On the content page...
  a. Leave the Stacked Content editor as empty
  b. Try to publish to get a a validation error
  c. Add a new item to Stacked Content
  d. Press the "Submit" button... check the JS console for the error.

It seems that the validation error has set the value of `$scope.model.value` to and empty string, so it is no longer an array. 😕 

As a workaround, I've added in the following code, (around line 97)

```js
if (!($scope.model.value instanceof Array)) {
    $scope.model.value = [];
}
```

@mattbrailsford Does that seem an okay fix?
